### DISCUSSION TITLE
Linking to RTL simulator

### DISCUSSION THREAD

**Discussion by sequencer** [sparcians/map#384] [General]
I want to use Sparta as a BFM(Bus Functional Model) framework for SoC generator. I wonder if it is possible to do so, and have a question about how to linking to RTL simulator for RTL+modeling cosim.

**Comment by ghost** [comment]
Definitely you can use Sparta to build a BFM -- been done before!  Connecting it to existing models like RTL or SystemC has also been done before using TLM interfaces -- this is something that SystemC has good support for.

Sparta has examples of how to do this, but it requires building Sparta with SystemC "awareness".  Follow the directions [here](https://sparcians.github.io/map/systemc.html) to build Sparta with SystemC support and to build the Sparta TLM [examples](https://github.com/sparcians/map/tree/master/sparta/example/SystemC).

The TLM examples show how you can use SystemC's ability to work with RTL and connect it to Sparta.  The TLM examples are made of 2 parts:

1. The TLM examples ([part of the SystemC download](https://github.com/accellera-official/systemc/tree/master/examples))
2. The [Sparta SystemC/TLM Gasket](https://github.com/sparcians/map/blob/master/sparta/example/SystemC/common/sparta-target/SpartaTLMTargetGasket.hpp) to connect the examples to a dummy Sparta memory model

If I were to embark on the task you are heading down, I'd look at TLM interfaces from RTL to hook to my Sparta based model.  Then, using the [SysCSpartaSchedulerAdapter](https://sparcians.github.io/map/classsparta_1_1SysCSpartaSchedulerAdapter.html), hook my Sparta model up to the RTL/SysC model.

Hope this helps.
